### PR TITLE
Tickets/dm 35914

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

**Review by r-owen — APPROVED** [review]
Looks good. I'm sorry it's still so verbose to generate a test, but it still looks like an improvement to me. 

One suggestion I didn't think of until now is to rename "perform_amcs_motion_test" to "validate_amcs_motion" or similar -- i.e. always use the same prefix ("validate_" in your case) to name any method that does some kind of generic checking.

File: tests/test_amcs.py [hunk 1]
```diff
@@ -36,12 +36,24 @@
 )
 
 # The maximum AZ rotation speed (rad/s)
-MAX_SPEED = math.radians(4.0)
+MAX_SPEED = 4.0
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply tests/test_amcs.py:39]: Thanks, yes of course. Done.

File: tests/test_amcs.py [hunk 2]
```diff
@@ -36,12 +36,24 @@
...
 START_TAI = 10001.0
 # The amount of time needed before the motors of the AZ rotation begin moving.
 START_MOTORS_ADD_DURATION = 5.5
 
 
+class ExpectedState:
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply tests/test_amcs.py:45]: Great suggestion. Done.

File: tests/test_amcs.py [hunk 3]
```diff
@@ -101,521 +113,411 @@ async def verify_amcs(
...
+        duration = await self.amcs.crawlAz(
+            velocity=math.radians(crawl_velocity),
+            start_tai=start_tai,
+        )
+        assert pytest.approx(duration) == expected_duration
+
+    async def verify_halt(
+        self,
+        start_tai: float,
+        expected_states: dict[float, ExpectedState],
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply tests/test_amcs.py:145]: Another great suggestion. Done.

**Pull request merged**